rust/foxglove/src/websocket.rs

```diff
@@ -887,6 +977,132 @@ impl Server {
                 client.addr
             );
         }
+
+        if !services.is_empty() {
+            let msg = Message::text(protocol::server::advertise_services(
+                services.iter().map(|s| s.as_ref()),
+            ));
+            if let Err(err) = sender.send(msg).await {
+                tracing::error!("Error advertising services: {err}");
+            } else {
+                for service in services {
+                    tracing::info!(
+                        "Advertised service {} with id {} to client {}",
+                        service.name(),
+                        service.id(),
+                        client.addr
+                    );
+                }
+            }
+        }
+    }
+
+    /// Adds new services, and advertises them to all clients.
+    ///
+    /// This method will fail if the services capability was not declared, or if a service name is
+    /// not unique.
+    pub async fn add_services(&self, new_services: Vec<Service>) -> Result<(), FoxgloveError> {
```

Comment:
This doesn't need to be async. The only thing async it calls is control_plane_tx.send_async, which only suspends when the queue is full. You can use send_control_msg and slow clients will get disconnected. Then both the blocking and non-blocking handles can call it directly.

We'll make the queue depths configurable in the future so the user can adjust that if need be.